src/DotLiquid/Tags/Decrement.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using DotLiquid.Exceptions;
+using DotLiquid.Util;
+
+namespace DotLiquid.Tags
+{
+    /// <summary>
+    /// Decrements a computed value
+    /// </summary>
+    public class Decrement : Tag
+    {
+        private static readonly Regex Syntax = R.B(R.Q(@"(?<Variable>{0}+)\s*"), Liquid.VariableSignature);

```

Comment:
Variable signature includes dotted and bracketed notations. This should be a simple variable name imho. See capture block, but keep in mind to fix the issue reported in #331 .